lib/vanagon/platform/solaris_10.rb

```diff
@@ -42,14 +42,16 @@ def generate_package(project)
           # - The bin directory and all bin files are owned by root:bin instead of root:sys
           # - All files under lib are owned by root:bin instead of root:sys
           # - All .so files are owned by root:bin instead of root:sys
+          # - Explicity only include puppet directories 
+          #   (this should exclude things like root/bin root/var and such)
           %((cd $(tempdir)/#{name_and_version}; pkgproto . | sort | awk ' \
-            $$1 ~ /^d$$/ {print "d",$$2,$$3,"0755 root sys";} \
+            $$1 ~ /^d$$/ && $$3 ~ /^.*puppet.*/ {print "d",$$2,$$3,"0755 root sys";} \
```

Comment:
As this is a generic application, non-specific to puppet, it would be better to not special case puppet here.

Vanagon projects already register directories that should be included in the packages, so maybe that would be useful here.